File: racingcar/racingcar/domain/NumberGenerator/RandomNumberGenerator.swift
//
//  RandomNumberGenerator.swift
//  racingcar
//
//  Created by KimMinSeok on 4/4/24.
//

import Foundation

class RandomNumberGenerator : NumberGeneator{
    func getNumber() -> Int {
        return Int.random(in: MIN_RANDOM_NUMBER...MAX_RANDOM_NUMBER)
    }
    
    private let MIN_RANDOM_NUMBER = 0
    private let MAX_RANDOM_NUMBER = 9
}

File: src/main/kotlin/racingcar/domain/numbergenerator/RandomNumberGenerator.kt
package racingcar.domain.numbergenerator

class RandomNumberGenerator : NumberGenerator {
    override fun getNumber(): Int {
        return (MIN_RANDOM_NUMBER..MAX_RANDOM_NUMBER).random()
    }

    companion object {
        private const val MIN_RANDOM_NUMBER = 0
        private const val MAX_RANDOM_NUMBER = 9
    }
}
